src/ui/base-page-storyEdit.js

```diff
@@ -208,6 +251,89 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         }
     });
 
+    /**
+     * Saves story content to a given key in the browser's localStorage object.
+     * Since localStorage can only store strings, the content must be serialized.
+     *
+     * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage}
+     *
+     * @param {String} storyAutosaveKey - the key at which to save the story content
+     * @param {Object} storyContent - the story content (model data) to save
+     */
+    sjrk.storyTelling.base.page.storyEdit.saveStoryToAutosave = function (storyAutosaveKey, storyContent) {
+        try {
+            // clear the previously-saved story before saving the current story
+            sjrk.storyTelling.base.page.storyEdit.clearAutosave(storyAutosaveKey);
+
+            var serialized = JSON.stringify(storyContent);
+            window.localStorage.setItem(storyAutosaveKey, serialized);
+        } catch (ex) {
+            fluid.log(fluid.logLevel.WARN, "An error occurred when saving", ex);
+        }
+    };
+
+    /**
+     * Loads story content from a given key in the browser's localStorage object.
+     * Once the content is loaded, the storyEditor will be updated to match.
+     *
+     * @param {String} storyAutosaveKey - the key to load the story content from
+     * @param {Component} storyEditor - an instance of `sjrk.storyTelling.ui.storyEditor`
+     * @param {String|String[]} storyPath - the model path to load the story to
+     * @param {String} sourceName - the name of the Infusion change source for this update
+     */
+    sjrk.storyTelling.base.page.storyEdit.loadStoryFromAutosave = function (storyAutosaveKey, storyEditor, storyPath, sourceName) {
+        try {
+            // localStorage can only store string values
+            var savedStory = JSON.parse(window.localStorage.getItem(storyAutosaveKey));
+
+            if (savedStory) {
+                // media block blob URLs are no longer valid/reliable after a page reload
+                sjrk.storyTelling.base.page.storyEdit.clearMediaBlockUrls(savedStory.content);
```

Comment:
I think it's better to clear media block urls before the save rather than at reading back the saved.

Reply:
If I clear the block URL's on saving, then they will be nulled out while the user is still editing which will cause unexpected behaviour (e.g. images will look right but not publish). I think it's better to do it on load, since we know that's a recovery situation and it's okay to clear them.

Reply:
A reasonable point and I agree.